livekit-agents/livekit/agents/voice/room_io/_input.py

```diff
@@ -198,6 +214,14 @@ def _on_track_unavailable(
             if self._on_track_available(publication.track, publication, participant):
                 return
 
+    def _on_token_refreshed(self) -> None:
+        if (
+            self._processor is not None
+            and self._room._token is not None
+            and self._room._server_url is not None
+        ):
+            self._processor._update_credentials(token=self._room._token, url=self._room._server_url)
```

Comment:
Do you think the livekit-rtc `package` should be responsible for directly updating the credentials?

Reply:
agree, in theory that would be nice.

I wasn't sure what the best way would be for handling the possibility of the input stream changing here further down in the rtc SDK. More than the credential update this would be about the stream info update.
 
But that makes me realize the current approach is also flawed when that happens because the processor is still attached to the "old" track when the input changes, we'd have to stop it on the old one.

Any thoughts on this?